src/SqlAsyncCollector.cs

```diff
@@ -95,7 +96,19 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
                 TelemetryInstance.TrackEvent(TelemetryEventName.AddAsync);
                 try
                 {
-                    this._rows.Add(item);
+                    if (item is string)
```

Comment:
So for these it seems like T might be a string or a JObject? It would be clearer to compare T directly instead of the item IMO (since the item is always type `T` per the signature typing)

Reply:
I tries using T and compiler won't let me -> 'T' is a type, which is not valid in the given context

Reply:
You have to do `typeof(T) == typeof(JObject)`